internal/p2p/shim.go

```diff
@@ -1,58 +1,42 @@
 package p2p
 
 import (
-	"errors"
 	"sort"
 
 	"github.com/gogo/protobuf/proto"
+	"github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/internal/p2p/conn"
 	"github.com/tendermint/tendermint/libs/log"
 )
 
-// ============================================================================
-// TODO: Types and business logic below are temporary and will be removed once
-// the legacy p2p stack is removed in favor of the new model.
-//
-// ref: https://github.com/tendermint/tendermint/issues/5670
-// ============================================================================
-
-var _ Reactor = (*ReactorShim)(nil)
-
-type (
-	messageValidator interface {
-		Validate() error
-	}
-
-	// ReactorShim defines a generic shim wrapper around a BaseReactor. It is
-	// responsible for wiring up legacy p2p behavior to the new p2p semantics
-	// (e.g. proxying Envelope messages to legacy peers).
-	ReactorShim struct {
-		BaseReactor
-
-		Name        string
-		PeerUpdates *PeerUpdates
-		Channels    map[ChannelID]*ChannelShim
-	}
+// ChannelDescriptorShim defines a shim wrapper around a legacy p2p channel
```

Comment:
Should we use "Legacy" in the name? Also possibly document that this is intended to go away, though if this is just the first step of more to come, I'm not deeply worried about either of those.

Reply:
yeah, this is just transient, the shims and channel registration will get deleted in the second pass, but it needs a little more fineness than I wanted to do in this PR. just to be clear: this file goes away, and channel registration will get ver